File: packages/shared-types/src/routine.ts
export type RoutineCategory =
  | 'cooking'
  | 'hygiene'
  | 'laundry'
  | 'medication'
  | 'transit'
  | 'shopping'
  | 'cleaning'
  | 'social'
  | 'custom';

export type ScheduleType = 'daily' | 'weekly' | 'custom' | 'on_demand';

export interface ScheduleConfig {
  days?: string[]; // ['mon', 'tue', ...]
  time?: string; // '08:00'
  interval?: number; // days between
}

export interface Routine {
  id: string;
  title: string;
  description?: string;
  category: RoutineCategory;
  icon?: string;
  coverImageUrl?: string;
  createdBy: string;
  assignedTo?: string;
  isTemplate: boolean;
  complexity: number; // 1-5
  estimatedMinutes?: number;
  isActive: boolean;
  scheduleType?: ScheduleType;
  scheduleConfig?: ScheduleConfig;
  steps?: RoutineStep[];
  createdAt: string;
  updatedAt: string;
}

export interface RoutineStep {
  id: string;
  routineId: string;
  stepOrder: number;
  title: string;
  instruction?: string;
  imageUrl?: string;
  videoUrl?: string;
  audioUrl?: string;
  audioTtsText?: string;
  durationHint?: number; // seconds
  requiresConfirmation: boolean;
  checkpoint: boolean;
  instructionSimple?: string;
  instructionDetailed?: string;
  createdAt: string;
  updatedAt: string;
}

File: apps/neuronav_app/lib/data/models/routine.dart
import 'package:json_annotation/json_annotation.dart';

part 'routine.g.dart';

@JsonSerializable(fieldRename: FieldRename.snake)
class StepRow {
  final String id;
  final int stepOrder;
  final String title;
  final String instruction;
  final String? instructionSimple;
  final String? instructionDetailed;
  final String? imageUrl;
  final String? audioUrl;
  final String? videoUrl;
  final int durationHint;
  final bool checkpoint;

  const StepRow({
    required this.id,
    required this.stepOrder,
    required this.title,
    required this.instruction,
    this.instructionSimple,
    this.instructionDetailed,
    this.imageUrl,
    this.audioUrl,
    this.videoUrl,
    this.durationHint = 60,
    this.checkpoint = false,
  });

  factory StepRow.fromJson(Map<String, dynamic> json) =>
      _$StepRowFromJson(json);

  Map<String, dynamic> toJson() => _$StepRowToJson(this);
}

@JsonSerializable(fieldRename: FieldRename.snake)
class RoutineRow {
  final String id;
  final String title;
  final String? description;
  final String category;
  final bool isActive;
  final String? assignedTo;
  final int? complexityLevel;
  final List<StepRow>? routineSteps;
  final String? createdAt;

  const RoutineRow({
    required this.id,
    required this.title,
    this.description,
    required this.category,
    this.isActive = true,
    this.assignedTo,
    this.complexityLevel,
    this.routineSteps,
    this.createdAt,
  });

  factory RoutineRow.fromJson(Map<String, dynamic> json) =>
      _$RoutineRowFromJson(json);

  Map<String, dynamic> toJson() => _$RoutineRowToJson(this);
}
